### PR TITLE
Added support to multiples versicles

### PR DESCRIPTION
**Pull request by JoaoEmanuell**
In this pull request, I am looking to add support for multiple verses separated into a single text.

Currently, the plugin only supports initial and final verses, such as: `John 5:1-3`. With this addition, the plugin now allows verses to be separated, such as: `John 5:1-3, 8, 12-15`, which improves usability and content generation.

However, after implementation, the only format accepted now is `book chapter:verses`, replacing the previous format that allowed `book chapter, verses`.

In addition, changes were made to the names of the books in Brazilian Portuguese, removing the accents to ensure easier book search.

The plugin was tested on Linux and Android platforms, both using Obsidian version 1.6.7.

### CODE REVIEW

File: src/EditorSuggester.ts [hunk 1]
```diff
@@ -26,9 +26,8 @@ export class EditorSuggester extends EditorSuggest<VerseLink> {
 		cursor: EditorPosition,
 		editor: Editor,
 		file: TFile | null
-	): EditorSuggestTriggerInfo | null {
+	): EditorSuggestTriggerInfo | null {
```
Comment (jaanonim) [inline src/EditorSuggester.ts:29]: Please fix those weird trailing whitespace :-)

File: src/EditorSuggester.ts [hunk 2]
```diff
@@ -116,14 +115,12 @@ export function getSuggestionsFromQuery(
 		return [];
 	}
 
-	const numbersPartsOfQueryString = query.substring(bookName.length);
+	const numbersPartsOfQueryString = query.substring(bookName.length);		
 	const numbers = numbersPartsOfQueryString.split(separatorRegex);
-
+	const verses = numbersPartsOfQueryString.split(':')[1].replaceAll(' ', '').trim() // get verses provide by user
```
Comment (jaanonim) [inline src/EditorSuggester.ts:120]: This throws when trying to link whole chapter.
```
EditorSuggester.ts:120 Uncaught 
TypeError: Cannot read properties of undefined (reading 'replaceAll')
    at getSuggestionsFromQuery (EditorSuggester.ts:120:55)
```

File: src/Verse.ts [hunk 3]
```diff
@@ -7,8 +7,7 @@ export default abstract class Verse {
 		private bookUrl: string,
 		private book: string,
 		private chapter: number,
-		private verse: number | undefined,
-		private verseEnd: number | undefined
+		private verses: string
```
Comment (jaanonim) [inline src/Verse.ts:10]: We cannot store verses as string entered by user - it breaks with en-dash support and other chapter separators like `,` and `.`. I suggest using list of objects like:
```ts
{
    start: number,
    end: number | undifined
}
```

File: tsconfig.json [hunk 4]
```diff
@@ -17,7 +17,8 @@
       "DOM",
       "ES5",
       "ES6",
-      "ES7"
+      "ES7",
+      "ES2021.String"
```
Comment (jaanonim) [inline tsconfig.json:21]: Is "ES2021.String" needed?

**Review by jaanonim — CHANGES_REQUESTED** [review]
Thanks you for contributing, When I was making this project I didn't expected that so many people will find it useful and will want to contribute logic code.

1. It is not acceptable for me to drop support for `book chapter, verses` notation (I'm form Poland and here it's the most popular notation eg. `2 Kor 1, 1-2.10` = `2 Co 1:1-2, 10`). Here is [refrence](https://github.com/jaanonim/obsidian-youversion-linker/issues/12#issuecomment-1880053805) for you.

2. It breaks support for whole chapter linking like `John 5`.

3. The plugin supports both normal dash `-` and en-dash `—`, but this change breaks support for en-dash - it get processed but invalid link is generated.

To sum up I would love to merge that future, but there need to be some changes to make it compatible with every use case.

File: tsconfig.json (hunk 4 above)
Comment (jaanonim) [reply tsconfig.json:21]: Ok I see it's need for that [`replaceAll`](https://github.com/jaanonim/obsidian-youversion-linker/pull/32/files#diff-f9f01c50435a34208cf5fcdf172fd8c4c324da6cc09522cd9a695e4be9a648e1R120) but I don't think that `replaceAll` will stay there as it throws errors.

**Comment by JoaoEmanuell** [comment]
Hello, thanks for the suggestions, I managed to add support for texts with dots, commas, and en-dash.

I tested them in Brazilian Portuguese, English, and Polish and they all work normally.

Now the plugin supports texts with the structure: `John 5: 1-2,6` | `John 5.1-2,6` and `John 5,1-2,6—9`, the links are generated correctly and are not broken during generation.

However, I was unable to add support for the entire chapter (`John 5`) even in the original plugin I cannot insert the chapter this way, it only recognizes them when I insert the verses.

**Review by jaanonim — APPROVED** [review]

**Comment by jaanonim** [comment]
Ok, I have made some changes myself and merege it. Ther will be new version in a moment. If you will find any bugs feel free to open issue.
